kaggle_environments/envs/llm_20_questions/llm_20_questions.py

```diff
@@ -34,8 +35,19 @@
 RANDOM_GUESSER = "random_guesser"
 RANDOM_ANSWERER = "random_answerer"
 
+def weighted_random_category(keywords_list):
+    cat_words = [[cat_entry, len(cat_entry["words"])] for cat_entry in keywords_list]
+    total_weight = sum([entry[1] for entry in cat_words])
+    cat = math.floor(random.random() * total_weight)
+    total = 0
+    for entry in cat_words:
+        total = total + entry[1]
+        if total >= cat:
+            return entry[0]
+    return random.choice(keywords_list)
```

Comment:
Not 100% sure but this looks good, but IMO better to use a library.  Looks like `random.choices` does exactly what you're looking for, here's an example from an unnamed LLM:
```
elements = ["apple", "banana", "orange"]
weights = [0.5, 0.2, 0.3]  # Probabilities for each element

chosen = random.choices(elements, weights=weights, k=1)[0]  
print(chosen)  # Outputs one of the fruits, more likely "apple"
```